addons/stock/models/stock_quant.py

```diff
@@ -935,8 +935,7 @@ def _onchange_location_or_product_id(self):
             quant = self._gather(
                 self.product_id, self.location_id, lot_id=self.lot_id,
                 package_id=self.package_id, owner_id=self.owner_id, strict=True)
-            if quant:
-                self.quantity = sum(quant.filtered(lambda q: q.lot_id == self.lot_id).mapped('quantity'))
+            self.quantity = sum(quant.filtered(lambda q: q.lot_id == self.lot_id).mapped('quantity')) if quant else 0
```

Comment:
```suggestion
            self.quantity = sum(quant.filtered(lambda q: q.lot_id == self.lot_id).mapped('quantity'))
```
The `filtered()` will return an empty recordset that will have a `quantity` of 0.